### PR TITLE
DM-8833: Creates a basic download dialog based on selected rows

### PR DESCRIPTION
**Pull request by loitly**
https://jira.lsstcorp.org/browse/DM-8833
 - create download dialog with extendable options
 - create download button that connects to single table or active table of a named group
 - update LcResults.jsx to use this new download dialog as an example

To test:
http://localhost:8080/firefly/lc.html
upper right corner 'Prepare Download' button is tied to active table.
select one or more rows, then click 'Prepare Download' to show the dialog.

### CODE REVIEW

File: src/firefly/js/templates/lightcurve/LcResult.jsx [hunk 1]
```diff
@@ -155,7 +152,27 @@ const StandardView = ({visToolbar, title, searchDesc, standard, imagePlot, xyPlo
...
+                                    MaxBundleSize: 200*1024*1024,    // set it to 200mb to make it easier to test multi-parts download.  each wise image is ~64mb
+                                    FilePrefix: 'WISE_Files',
+                                    BaseFileName: 'WISE_Files',
+                                    DataSource: 'WISE images',
+                                    FileGroupProcessor: 'LightCurveFileGroupsProcessor'
+                                }}>
+                                <ValidationField
+                                        initialState= {{
+                                               value: 'A sample download',
+                                               label : 'Title for this download:'
```
Comment (tgoldina) [inline src/firefly/js/templates/lightcurve/LcResult.jsx]: Is it possible to make the default title more descriptive, for example, by including the type (cutout or image) and the number of files to be downloaded?

File: src/firefly/js/ui/DownloadDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        BaseFileName:   PropTypes.string,
+        Title:          PropTypes.string,
+        DataSource:     PropTypes.string,
+        MaxBundleSize:  PropTypes.number,
+        FileGroupProcessor: PropTypes.string.isRequired,
+    })
+};
+
+DownloadOptionPanel.defaultProps= {
+    groupKey: 'DownloadDialog',
```
Comment (tgoldina) [inline src/firefly/js/ui/DownloadDialog.jsx]: Just noticed all these trailing commas. Looks like Trey finally moved you on his side. :)

File: src/firefly/js/ui/FormPanel.jsx [hunk 3]
```diff
@@ -42,13 +43,21 @@ export const FormPanel = function (props) {
...
+                                    onSuccess={createSuccessHandler(action, params, onSubmit)}
+                                    onFail={onError || handleFailfure}
+                                    text = {submitText}
+                    />
+                    <button style={{display: 'inline-block'}} type='button' className='button std' onClick={onCancel}>Cancel</button>
+
+                </div>
+                <div>
+                    {help_id && <HelpIcon helpId={help_id} />}
+                </div>
```
Comment (tgoldina) [inline src/firefly/js/ui/FormPanel.jsx]: Nice addition.

File: src/firefly/js/templates/lightcurve/LcResult.jsx (hunk 1 above)
Comment (loitly) [inline src/firefly/js/templates/lightcurve/LcResult.jsx]: This is throw-away code.  It serves as an example as well as for testing.

File: src/firefly/js/ui/DownloadDialog.jsx (hunk 2 above)
Comment (loitly) [inline src/firefly/js/ui/DownloadDialog.jsx]: yes.. my lazy side wins.

**Pull request merged**